src/esm_runscripts/prepare.py

```diff
@@ -672,6 +674,50 @@ def set_parent_info(config):
     return config
 
 
+def add_vcs_info(config):
+    """
+    Adds version control system information in a plain text yaml file under the
+    experiment log directory for both the models and the esm-tools.
+
+    Parameters
+    ----------
+    config : dict
+        The experiment configuration
+
+    Returns
+    -------
+    config : dict
+        The experiment configuration
+    """
+    import pdb
+    pdb.set_trace()
+    exp_vcs_info_file = (
+        f"{config['general']['thisrun_log_dir']}/{config['general']['expid']}_vcs_info.yaml"
+    )
+    logger.debug("Experiment information is being stored for usage under:")
+    logger.debug(f">>> {exp_vcs_info_file}")
+    vcs_versions = {}
+    all_models = config.get("general", {}).get("models", [])
+    for model in all_models:
+        model_dir = config[model]["model_dir"]
+        if helpers.is_git_repo(model_dir):
+            try:
+                vcs_versions[model] = helpers.get_git_hash
+            except helpers.GitDirtyError:
+                vcs_versions[model] = "unknown (dirty repo detected)"
+
+    # NOTE(PG): There is no good way to get the repo directory from the config,
+    # this may at least be a good start:
+    esm_tools_repo = config.get("general", {}).get("esm_function_dir")
+    if esm_tools_repo is not None:
+        vcs_versions["esm_tools"] = helpers.get_git_hash(f"{esm_tools_repo}/../")
```

Comment:
As I wrote to the function definition, this function raises and exception and it will terminate the flow. Maybe it needs to be placed in a try-catch block like above.